src/proto/generation.proto

```diff
@@ -267,6 +268,93 @@ message ClassifierParameters {
   optional Action realized_action = 3;
 }
 
+
+//
+// Interpolation
+//
+
+enum InterpolateMode {
+  INTERPOLATE_LINEAR = 0;
+  INTERPOLATE_RIFE = 1;
+  INTERPOLATE_VAE_LINEAR = 2;
+  INTERPOLATE_VAE_SLERP = 3;
+}
+
+// Interpolation between two images applied at specified blend ratios
+message InterpolateParameters {
+  repeated float ratios = 1;
+  optional InterpolateMode mode = 2;
+}
+
+
+//
+// Transforms
+//
+
+enum BorderMode {
+  BORDER_REFLECT = 0;   // reflect image values across the border
+  BORDER_REPLICATE = 1; // replicate border values outside the image
+  BORDER_WRAP = 2;      // wrap around / tile the image values
+  BORDER_ZERO = 3;      // use 0 for locations outside the image
+}
+
+enum ColorMatchMode {
+  COLOR_MATCH_HSV = 0;  // match hue, saturation, and value histograms
```

Comment:
If we don't standardize on reserving 0 for the null case (i.e. something like `COLOR_MATH_NONE = 0;`) then we're likely committing to having the specific values used by the grpc protocol being defined differently than elsewhere.. since it's possible that in either a file format or client implementation somewhere, we're going to want to use an actual value for the color match mode all the time instead of just omitting it when it's the null case. This can be fine, but I want to make sure we're making a conscious decision to generally not care whether or not the grpc protocol values are the same as the ones used in other contexts.

Reply:
Never mind - I think it's pretty standard to not be concerned because we can map it. I'm just being weird.